.github/workflows/build-ubuntu.yml

```diff
@@ -34,7 +33,17 @@ jobs:
       fail-fast: true
       max-parallel: 4
       matrix:
-        TAG: [latest, 20.04, 18.04, 16.04]
+        include:
+          - TAG: latest
+            PLATFORMS: linux/amd64,linux/arm64
+          - TAG: 20.04
+            PLATFORMS: linux/amd64,linux/arm64
+          - TAG: 18.04
+            PLATFORMS: linux/amd64
+          - TAG: 16.04
+            PLATFORMS: linux/amd64
+    env:
+      PLATFORMS: ${{ matrix.PLATFORMS }}
     steps:
       - name: Login to GitHub Container Registry
         if: ${{ github.repository_owner == github.actor }}
```

Comment:
I think what you may want here is instead a check if this is a pull request or not. When you make a pull request, the github secret's won't be available, but if you merge something to a branch or you manually trigger a workflow (workflow_dispatch) the secrets will be available and a build/push could work.

Perhaps something that checks the event type for not being a pull_request would make sense instead?

Reply:
You may be the actor of a workflow execution triggered via a pull_request event, and then the secrets won't be available even though you are the actor and owner.

Reply:
I did that before and don't want to now since I create pull requests too

Reply:
Unfortunately this is a problem I hate GitHub for since it affects this repo and nektos/act. It would be easy to solve and not expose secrets completely but GitHub is [censored]. :>

Reply:
Although this specific workflow run might be because I had to approve the run.

Reply:
I think I miss out on what you mean.

I do feel good about GitHub not exposing `${{ secrets.QUAY_TOKEN }}` for example when executing a workflow triggered by a pull_request as compared to being merged (push event) or explicitly dispatched (workflow_dispatch event). I'm thinking that projects I work on could receive a PR that would be modifying the workflows to expose such secrets or similar otherwise.

Reply:
Thanks for merge, I'd be very happy to follow this up with help down the line if there is trouble. I'll go ahead and watch this repo.

Reply:
> I think I miss out on what you mean.
> 
> I do feel good about GitHub not exposing `${{ secrets.QUAY_TOKEN }}` for example when executing a workflow triggered by a pull_request as compared to being merged (push event) or explicitly dispatched (workflow_dispatch event). I'm thinking that projects I work on could receive a PR that would be modifying the workflows to expose such secrets or similar otherwise.

Sure, it's very useful and secure, but there are cases where I would like to expose the token to workflow. example: https://github.com/nektos/act/pull/773

If someone were to maliciously steal it they would hit the block like your PR, where I have to allow the workflow run :)